### PR TITLE
Cherry pick `bugfix/rich-text-composer-not-work` (PR #25)

### PR DESCRIPTION
**Pull request by dab246**
## Dependent

- Need merged: https://github.com/linagora/enough_html_editor/pull/31

## Desc

- Cherry pick PR #25

### CODE REVIEW

File: lib/views/widgets/divider/custom_horizontal_divider.dart [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+import 'package:flutter/material.dart';
+import 'package:rich_text_composer/views/commons/colors.dart';
+
+class CustomHorizontalDivider extends StatelessWidget {
```
Comment (tddang-linagora) [inline lib/views/widgets/divider/custom_horizontal_divider.dart:5]: Can `Divider` widget replace this?

File: lib/views/widgets/divider/custom_vertical_divider.dart [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
+import 'package:flutter/material.dart';
+import 'package:rich_text_composer/views/commons/colors.dart';
+
+class CustomVerticalDivider extends StatelessWidget {
```
Comment (tddang-linagora) [inline lib/views/widgets/divider/custom_vertical_divider.dart:5]: Can `VerticalDivider` widget replace this?

File: lib/views/widgets/divider/custom_horizontal_divider.dart (hunk 1 above)
Comment (dab246) [reply lib/views/widgets/divider/custom_horizontal_divider.dart:5]: No, we need custom it. Divider is not rendering as we expected

File: lib/views/widgets/divider/custom_vertical_divider.dart (hunk 2 above)
Comment (dab246) [reply lib/views/widgets/divider/custom_vertical_divider.dart:5]: No, we need custom it. VerticalDivider is not rendering as we expected

File: android/src/main/kotlin/flutter/plugin/rich_text_composer/KeyboardUtils.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package flutter.plugin.rich_text_composer
+
+import android.content.Context
+import android.view.inputmethod.InputMethodManager
+object KeyboardUtils {
+    fun showSoftInput(context: Context) {
+        val inputMethodManager =
+            context.getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
+        inputMethodManager?.toggleSoftInput(
```
[the remark this answers is not in the document]
Comment (dab246) [reply android/src/main/kotlin/flutter/plugin/rich_text_composer/KeyboardUtils.kt:9]: According to android documentation `toggleSoftInput` is deprecated on Android 12. And replaced by `showSoftInput(android.view.View, int)` and `hideSoftInputFromWindow(android.os.IBinder, int)`. But if `showSoftInput` is used, it requires a specific `View` to be passed in. We cannot (have not found a way yet) reference a `widget` from flutter dart code to `View` of native code. On the other hand, `toggleSoftInput` is only marked as deprecated but has not been `removed`, so we can still use it at this time.

File: android/src/main/kotlin/flutter/plugin/rich_text_composer/KeyboardUtils.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+            InputMethodManager.SHOW_FORCED,
```
[the remark this answers is not in the document]
Comment (dab246) [reply android/src/main/kotlin/flutter/plugin/rich_text_composer/KeyboardUtils.kt:10]: idem

**Review by tddang-linagora — APPROVED** [review]

File: pubspec.yaml [hunk 5]
```diff
@@ -10,13 +10,14 @@ environment:
...
-  flutter_keyboard_visibility: ^5.4.0
+  flutter_keyboard_visibility: ^6.0.0
   flutter_svg: ^2.0.2
+  flutter_portal: ^1.1.3
 
   enough_html_editor:
     git:
       url: https://github.com/linagora/enough_html_editor.git
-      ref: cnb_supported
+      ref: cherry-pick-fix-rich-text-format-style
```
Comment (hoangdat) [inline pubspec.yaml:20]: please update this

Comment (dab246) [reply pubspec.yaml]: done

**Pull request merged**